csi/util.go

```diff
@@ -239,6 +242,38 @@ func parseJSONRecurringJobs(jsonRecurringJobs string) ([]longhornclient.Recurrin
 	return recurringJobs, nil
 }
 
+func syncMountPointDirectory(targetPath string) error {
+	d, err := os.OpenFile(targetPath, os.O_SYNC, 0750)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+
+	if _, err := d.Readdirnames(1); err != nil {
+		if !errors.Is(err, io.EOF) {
+			return err
+		}
+	}
+
+	// it would not always return `Input/Output Error` or `read-only file system` errors if we only use ReadDir() or Readdirnames() without an I/O operation
+	// an I/O operation will make the targetPath mount point invalid immediately
+	tempFile := targetPath + tempTestMountPointValidStatusFile
+	f, err := os.OpenFile(tempFile, os.O_CREATE|os.O_RDWR|os.O_SYNC|os.O_TRUNC, 0666)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			logrus.WithError(err).Warnf("Failed to close %v", tempFile)
+		}
+		if err := os.Remove(tempFile); err != nil {
+			logrus.WithError(err).Warnf("Failed to remove %v", tempFile)
+		}
+	}()
+
+	return f.Sync()
```

Comment:
NIT: Do we still need this after using `O_SYNC`?

Reply:
After some testing, it could not still detect the corrupted fs if there is no `f.Sync()` with the flag `O_SYNC`

Reply:
Looks like to detect a bad mount point, we cannot avoid `Sync()` so we will have to live with the fact that `Sync()` might take a long time when there is a lot of cache. So let's use it directly on the directory `d` instead. WDYT @shuo-wu @derekbit @mantissahz ? 

```
func syncMountPointDirectory(targetPath string) error {
	d, err := os.OpenFile(targetPath, os.O_SYNC, 0750)
	if err != nil {
		return err
	}
	defer d.Close()

	if _, err := d.Readdirnames(1); err != nil {
		if !errors.Is(err, io.EOF) {
			return err
		}
	}

	// it would not always return `Input/Output Error` or `read-only file system` errors if we only use ReadDir() or Readdirnames() without an I/O operation
	// an I/O operation will make the targetPath mount point invalid immediately
	return d.Sync()
```

Reply:
Testing that use `NewDirectFileIoProcessor` from `sparse-tools` libs without `Sync()` to open a file.

Reply:
It looks good to use `NewDirectFileIoProcessor` from sparse-tools libs to open/close/remove a file without `Sync()`.
A low chance will run to `WriteString` to trigger `Input/Output error`, almost it returns the `Input/Output error/read-only file system` errors at `Open`

Reply:
Use `WriteString` instead of `Sync()`